google/cloud/storage/quickstart/Makefile

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A simple Makefile to test the `install` target.
```

Comment:
We don't have a Makefile like this in `-spanner`, so I'm wondering if/why we need it here. Is this Makefile an example of something our *users* should do, or is this maybe more of a detail about how we'd like to implement a test. If the latter, it'd be nice if it didn't live in the quickstart/ directory; The point of quickstart/ is go give users a super-simple example to look at that should show them what *they* should do.

Is there anywhere else this could live?

Reply:
I should go and remove that line from the description...

> We don't have a Makefile like this in `-spanner`,

That is (arguably) a defect in `-spanner`.

> so I'm wondering if/why we need it here. Is this Makefile an example of something our _users_ should do, or is this maybe more of a detail about how we'd like to implement a test.

As you may recall, the most popular build systems for C++ are CMake, Visual Studio projects, and Make:

https://www.jetbrains.com/lp/devecosystem-2019/cpp/

Sadly we do not have an answer for Visual Studio projects, but we have an answer for CMake **and** Make.

> If the latter, it'd be nice if it didn't live in the quickstart/ directory; The point of quickstart/ is go give users a super-simple example to look at that should show them what _they_ should do.

If the users are based on Makefiles (the 3rd most popular option, and an order of magnitude more popular than Bazel), then this is how they would integrate the libraries into their build system.

> Is there anywhere else this could live?

I think `quickstart/` is exactly where it should be, we may be missing the corresponding section in the README.md file, but that could be a future improvement.

Reply:
OK, so then this is not just for our testing, rather it's an example for how our users would use this with `make`. In that case, I agree that it makes sense in `quickstart/`.

As you mentioned, we probably want to update the documentation comments in these quickstart/Makefile and quickstart/CMakeLists.txt files to no longer say that they are for testing something. Rather they are both simple examples for our users.